v2/metadata.go

```diff
@@ -84,32 +84,38 @@ func (s sanitizer) Sanitize(data interface{}) interface{} {
 	// Sanitizers are passed by value, so we can modify s and it only affects
 	// s.Seen for nested calls.
 	s.Seen = append(s.Seen, data)
+	t := reflect.TypeOf(data)
+	v := reflect.ValueOf(data)
+
+	if t == nil {
+		return "<nil>"
+	}
+
+	// Handle nil pointers and interfaces specifically
+	if t.Kind() == reflect.Interface || t.Kind() == reflect.Ptr {
```

Comment:
Is this enough for all cases? You can also have nil maps and slices. https://pkg.go.dev/reflect#Value.IsNil

Reply:
I guess it's the same behaviour as before at least, but would be good to know what happens for maps/slices...

Reply:
Added testcases for more types. For maps and slices the second switch case should catch it.